.github/workflows/documentation.yaml

```diff
@@ -1,5 +1,5 @@
 name: Docs
-on: [push, pull_request, workflow_dispatch]
+on: [pull_request, workflow_dispatch]
```

Comment:
Actually, you will need the `push` still because `push` is considered merging a pull request to `main` via a commit. 

A few issues I found:

1. The current `documentation.yaml` will rebuild the documentation every time, which overwrites the latest stable version -- set this job to only run when `push` is initiated ([example](https://github.com/E3SM-Project/e3sm_diags/blob/2b303dfdf1a83b115a8f372a81bf2f6b3bfbbc68/.github/workflows/build_workflow.yml#L113-L184) in e3sm_diags)
2. There is no versioning of documentation by branch and package version -- consider Read The Docs for this. e3sm_diags uses `sphinx-multiversion` which involves a complex setup process

Other suggestions:

1. You can combine `documentation.yaml` into `build_workflow.yaml` by creating a separate `publish-docs` job that only runs with `push` ([example](https://github.com/E3SM-Project/e3sm_diags/blob/2b303dfdf1a83b115a8f372a81bf2f6b3bfbbc68/.github/workflows/build_workflow.yml#L113-L184) in e3sm_diags)